### PR TITLE
[Prim] add meshgrid composite rule

### PR DESCRIPTION
**Pull request by rainyfly**
### PR types

Others
### PR changes

Ops
### Describe

add composite op meshgrid

### CODE REVIEW

File: python/paddle/incubate/autograd/composite_rules.py [hunk 1]
```diff
@@ -411,6 +411,30 @@ def silu_composite(x):
     return res
 
 
+@REGISTER_COMPOSITE('meshgrid')
+def meshgrid_composite(inputs):
+    """
+    define composite rule of op meshgrid
+    If the input has N tensors of size S_0, ... S_n-1, then the output will also have N tensors, where
+    each tensor is of shape (S_0, ..., S_n-1)
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py]: explain diff of each tensor

**Comment by cyber-pioneer** [comment]
image
make sure this code useful or add relative test case?

File: python/paddle/incubate/autograd/composite_rules.py [hunk 2]
```diff
@@ -411,6 +411,30 @@ def silu_composite(x):
...
+    """
+    size = len(inputs)
+    shape = [1 for i in range(size)]
+    for i in range(size):
+        dim = inputs[i].dim()
+        assert dim == 0 or dim == 1
+        if dim == 0:
+            shape[i] = 1
+        else:
+            shape[i] = inputs[i].shape[0]
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:460]: may the logic be simplified?

File: python/paddle/incubate/autograd/composite_rules.py (hunk 1 above)
Comment (rainyfly) [reply python/paddle/incubate/autograd/composite_rules.py]: done , add an example to demonstrate it

File: python/paddle/incubate/autograd/composite_rules.py (hunk 2 above)
Comment (rainyfly) [reply python/paddle/incubate/autograd/composite_rules.py:460]: Yes, this op can support zero dim tensor, so we should consider 0 dim tensor as shape [1]

File: python/paddle/incubate/autograd/composite_rules.py [hunk 3]
```diff
@@ -411,6 +411,35 @@ def silu_composite(x):
...
+    If the input has N tensors of size S_0, ... S_n-1, then the output will also have N tensors, where
+    each tensor is of shape (S_0, ..., S_n-1).
+    E.g. a1 is Tensor [1,2,3]
+         b1 is Tensor [4,5]
+         r1, r2 = paddle.meshgrid([a1, b1])
+         r1 is Tensor [[1,1], [2,2], [3,3]]
+         r2 is Tensor [[4,5], [4,5], [4,5]]
+    """
+    size = len(inputs)
+    shape = [1 for i in range(size)]
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py]: [1] * size

Comment (rainyfly) [reply python/paddle/incubate/autograd/composite_rules.py]: done

File: python/paddle/incubate/autograd/composite_rules.py [hunk 4]
```diff
@@ -411,6 +411,33 @@ def silu_composite(x):
...
+         b1 is Tensor [4,5]
+         r1, r2 = paddle.meshgrid([a1, b1])
+         r1 is Tensor [[1,1], [2,2], [3,3]]
+         r2 is Tensor [[4,5], [4,5], [4,5]]
+    """
+    size = len(inputs)
+    shape = [1] * size
+    for i in range(size):
+        dim = inputs[i].dim()
+        assert dim == 0 or dim == 1
```
Comment (Charles-hit) [inline python/paddle/incubate/autograd/composite_rules.py:458]: 这儿0-d相关的逻辑补充一下0-d的测试吧

Comment (rainyfly) [reply python/paddle/incubate/autograd/composite_rules.py:458]: done

File: python/paddle/fluid/tests/unittests/test_meshgrid_op.py [hunk 5]
```diff
@@ -28,6 +28,7 @@ def meshgrid_wrapper(x):
 class TestMeshgridOp(OpTest):
     def setUp(self):
         self.op_type = "meshgrid"
+        self.prim_op_type = "comp"
         self.python_api = meshgrid_wrapper
```
Comment (Charles-hit) [inline python/paddle/fluid/tests/unittests/test_meshgrid_op.py:32]: 看一下python3流水线 新增一下public_python_api

**Review by Charles-hit — APPROVED** [review]

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**